doc/source/train/user-guides/persistent-storage.rst

```diff
@@ -1,62 +1,370 @@
+        ...,
+        run_config=train.RunConfig(
+            storage_filesystem=fs,
+            storage_path="bucket-name/sub-path",
+            name="experiment_name",
+        )
+    )
+
+
+``fsspec`` filesystems
+~~~~~~~~~~~~~~~~~~~~~~~
 
-Many of these settings are configured through other config objects and passed through
-the ``RunConfig``. The following sub-sections contain descriptions of these configs.
+`fsspec <https://filesystem-spec.readthedocs.io/en/latest/>`_ offers many filesystem implementations,
+such as ``s3fs``, ``gcsfs``, etc.
 
-The properties of the run configuration are :ref:`not tunable <tune-search-space-tutorial>`.
+You can use any of these implementations by wrapping the ``fsspec`` filesystem with a ``pyarrow.fs`` utility:
 
-.. literalinclude:: ../doc_code/key_concepts.py
-    :language: python
-    :start-after: __run_config_start__
-    :end-before: __run_config_end__
+.. code-block:: python
+
+    # Make sure to install: `pip install -U s3fs`
+    import s3fs
+    import pyarrow.fs
+
+    s3_fs = s3fs.S3FileSystem(
+        key='miniokey...',
+        secret='asecretkey...',
+        endpoint_url='https://...'
+    )
+    custom_fs = pyarrow.fs.PyFileSystem(pyarrow.fs.FSSpecHandler(s3_fs))
+
+    run_config = RunConfig(storage_path="minio_bucket", storage_filesystem=custom_fs)
 
 .. seealso::
 
-    See the :class:`~ray.train.RunConfig` API reference.
+    See the API references to the ``pyarrow.fs`` wrapper utilities:
 
-    See :ref:`tune-storage-options` for storage configuration examples (related to ``storage_path``).
+    * https://arrow.apache.org/docs/python/generated/pyarrow.fs.PyFileSystem.html
+    * https://arrow.apache.org/docs/python/generated/pyarrow.fs.FSSpecHandler.html
 
 
-.. _train-log-dir:
 
-Persistent storage
-------------------
-Ray Train saves results and checkpoints at a persistent storage location.
-Per default, this is a local directory in ``~/ray_results``.
+MinIO and other S3-compatible storage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can follow the :ref:`examples shown above <custom-storage-filesystem>` to configure
+a custom S3 filesystem to work with MinIO.
+
+Note that including these as query parameters in the ``storage_path`` URI directly is another option:
 
-This default setup is sufficient for single-node setups or distributed
-training without :ref:`fault tolerance <train-fault-tolerance>`.
-When you want to utilize fault tolerance, require access to shared data,
-or are training on spot instances, it is recommended to set up
-a remote persistent storage location.
+.. code-block:: python
+
+    from ray import train
+    from ray.train.torch import TorchTrainer
+
+    trainer = TorchTrainer(
+        ...,
+        run_config=train.RunConfig(
+            storage_path="s3://bucket-name/sub-path?endpoint_override=http://localhost:9000",
+            name="experiment_name",
+        )
+    )
+
+
+Overview of Ray Train outputs
+-----------------------------
```

Comment:
Somewhere in this section it would be nice to include details on how the Results/Checkpoint attributes (e.g. path and filesystem) map back to persistent storage.